### ISSUE TITLE
[Bug]: `extra_kustomize_deployment_commands` seems to be required

### ISSUE THREAD

**Issue by FelixSab** [kube-hetzner/terraform-hcloud-kube-hetzner#852]
### Description

I've encountered a problem when using the kube-hetzner module with Terraform. Even though `extra_kustomize_deployment_commands` is stated to be an optional variable in the comments, I've received an error when I don't define it in my `kube.tf` file:

```
│ Error: remote-exec provisioner error
│ 
│   with module.kube-hetzner.null_resource.kustomization_user_deploy[0],
│   on .terraform/modules/kube-hetzner/kustomization_user.tf line 80, in resource "null_resource" "kustomization_user_deploy":
│   80:   provisioner "remote-exec" {
│ 
│ invalid empty string in 'scripts'
```

This error indicates that the variable `extra_kustomize_deployment_commands` is being used even when it's not defined, which leads to an error. I've managed to workaround this issue by defining the variable as follows:

```terraform
# Extra commands to be executed after the `kubectl apply -k` (useful for post-install actions, e.g. wait for CRD, apply additional manifests, etc.).
extra_kustomize_deployment_commands="echo"
```

### Kube.tf file

```terraform
locals {

  hcloud_token = "xxxxxxxxxxx"
}

module "kube-hetzner" {
  providers = {
    hcloud = hcloud
  }
  hcloud_token = var.hcloud_token != "" ? var.hcloud_token : local.hcloud_token


  source = "kube-hetzner/kube-hetzner/hcloud"


  ssh_public_key  = file("~/.ssh/id_ed25519.pub")
  ssh_private_key = file("~/.ssh/id_ed25519")


  network_region = "eu-central"
  control_plane_nodepools = [
    {
      name        = "control-plane-nbg1",
      server_type = "cpx11",
      location    = "nbg1",
      labels      = [],
      taints      = [],
    count = 1 }
  ]

  agent_nodepools = [
    {
      name                 = "internal-services",
      server_type          = "cx21",
      location             = "nbg1",
      labels               = ["internal", "production"],
      taints               = [],
      longhorn_volume_size = 0,
      count                = 1,
      backups              = true
    },
    {
      name                 = "production",
      server_type          = "cx21",
      location             = "nbg1",
      labels               = ["external", "production"],
      taints               = [],
      longhorn_volume_size = 0,
    count = 1 },
    {
      name                 = "database",
      server_type          = "cx31",
      location             = "nbg1",
      labels               = ["database"],
      taints               = [],
      longhorn_volume_size = 0,
    count = 1 }
  ]

  load_balancer_type     = "lb11"
  load_balancer_location = "nbg1"

  enable_wireguard = true


  enable_longhorn        = true
  longhorn_replica_count = 1

  # extra_kustomize_deployment_commands="echo"
}

provider "hcloud" {
  token = var.hcloud_token != "" ? var.hcloud_token : local.hcloud_token
}

terraform {
  required_version = ">= 1.3.3"
  required_providers {
    hcloud = {
      source  = "hetznercloud/hcloud"
      version = ">= 1.39.0"
    }
  }
}

output "kubeconfig" {
  value     = module.kube-hetzner.kubeconfig
  sensitive = true
}

variable "hcloud_token" {
  sensitive = true
  default   = ""
}
```

### Platform

WSL Ubuntu-22.04

**Comment by mysticaltech** [comment]
@FelixSab That seems to be a valid bug because it's a new feature that was not tested enough. Please don't hesitate to submit a PR with a fix, the help would be trully appreciated.

**Comment by FelixSab** [comment]
Done: #853

**Comment by kayiplar** [comment]
@FelixSab  @mysticaltech 
I am suggesting an additional change just in case the input is empty for any reason: https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/854  

This will avoid the error completely even when the variable is set to an empty string.

**Comment by mysticaltech** [comment]
Fixed in #854. Thanks again @FelixSab and @kayiplar.
